ethereum/contracts/BeefyLightClient.sol

```diff
@@ -287,24 +293,16 @@ contract BeefyLightClient {
         uint256 id,
         Commitment calldata commitment,
         ValidatorProof calldata validatorProof,
-        BeefyMMRLeaf calldata latestMMRLeaf,
-        SimplifiedMMRProof calldata proof
+        BeefyMMRLeaf calldata leaf,
+        SimplifiedMMRProof calldata leafProof
     ) public {
         verifyCommitment(id, commitment, validatorProof);
-        verifyNewestMMRLeaf(
```

Comment:
can we rather still do the verifyNewestMMRLeaf at the beginning, rather than within the  updateValidatorSet function? keeping up a check-effects-interactions pattern is still valuable for code readability

Reply:
Yeah, I was thinking the same. The code path in the current draft just helped my thinking around the solution. Will see how I can reorganize :+1: